### PR TITLE
[WIP][CODE CAMP] AbstractSampleChanger - Temp PR (added new test)

### PR DESCRIPTION
**Pull request by jbflo**

### CODE REVIEW

**Comment by jbflo** [comment]
I could not push to  mxcube/cc-AbstractSampleChanger 
Please @marcus-oscarsson have a look and move this PR to 502

File: test/pytest/test_sample_changer.py [hunk 1]
```diff
@@ -15,19 +10,32 @@ def test_sample_has_loaded_sample(beamline):
     assert(not beamline.sample_changer.has_loaded_sample())
 
 def test_sample_changer_unload(beamline):
-    pass
+    assert(beamline.sample_changer.unload() is not None)
```
Comment (marcus-oscarsson) [inline test/pytest/test_sample_changer.py:13]: Morning JB, Ill see you later today so this is just for later to not forget :)

You should perform a load first followed by an unload and then check for the presence of the sample you first loaded, like this you are more certain of the starting conditions.

File: test/pytest/test_sample_changer.py [hunk 2]
```diff
@@ -15,19 +10,32 @@ def test_sample_has_loaded_sample(beamline):
...
+       not beamline.sample_changer.has_loaded_sample() is None
+    ), "Sample changer  has_loaded_sample() must be  None (not initialized)"
+
 
 def test_sample_changer_select(beamline):
     pass
 
 def test_sample_changer_abort(beamline):
-    pass
+    assert(beamline.sample_changer.task_proc is not None)
```
Comment (marcus-oscarsson) [inline test/pytest/test_sample_changer.py:23]: Maybe even better is to start a load and then check the state (with get_state) after doing abort, the load has to be done with wait=False passed.

File: test/pytest/test_sample_changer.py [hunk 3]
```diff
@@ -15,19 +10,32 @@ def test_sample_has_loaded_sample(beamline):
...
+    beamline.sample_changer.abort()
+    assert(beamline.sample_changer.task_proc is None)
+
 
 def test_sample_changer_get_state(beamline):
```
Comment (marcus-oscarsson) [inline test/pytest/test_sample_changer.py:28]: You can improve this as well by for instance doing:

```
beamline.sample_changer.load((1, 1), wait=False)
assert(beamline.sample_changer. get_state() ==  SampleChangerState.BUSY)
beamline.sample_changer. wait_ready()
assert(beamline.sample_changer. get_state() ==  SampleChangerState.READY)
```

**Comment by marcus-oscarsson** [comment]
Hey JB, thanks !. I left a few comments (mostly for me not to forget) to be discussed when we meet later

File: HardwareObjects/abstract/AbstractSampleChanger.py [hunk 4]
```diff
@@ -681,7 +682,9 @@ def _unload(self, sample_slot=None):
         self._do_unload(sample_slot)
 
     def _resolve_component(self, component):
-        if component is not None and isinstance(component, basestring):
+        # The builtin basestring abstract type was removed. Use str instead.
+        # if component is not None and isinstance(component, basestring):
+        if component is not None and isinstance(component, str):
```
Comment (rhfogh) [inline HardwareObjects/abstract/AbstractSampleChanger.py:687]: That may give [prpoblems in Python 2.
Which is a pain, but as longa s we are still supporting it.
I can recommenf ConvertUtils.string_types, that takes care of the problem

**Comment by marcus-oscarsson** [comment]
Closing this one for now

**Pull request closed without merging**